lib/jxl/dec_external_image.cc

```diff
@@ -43,10 +44,12 @@ void FloatToU32(const float* in, uint32_t* out, size_t num, float mul,
     // instructions, so we run the non-SIMD path for those.
     vec_num = 0;
```

Comment:
I belive we should also remove vec_num; now it is either 0 or num, so it will be more readable, if we check `bits_per_sample == 32` around the places that depend on that.

Reply:
Sorry, didn't see this comment before; updated the patch with this change.